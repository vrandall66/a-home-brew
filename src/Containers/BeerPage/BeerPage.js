import React from 'react';

export const BeerPage = ({ beer }) => {
  return (
    <div>
      <h2>{beer.name}</h2>
      <h4>{beer.tagline}</h4>
      <aside>
        <h4>Ingredients</h4>
        <ul>
          Hops
          {beer.ingredients.hops}
        </ul>
        <ul>
          Malts
          {beer.ingredients.malt}
        </ul>
        <p>{beer.ingredients.yeast}</p>
      </aside>
    </div>
  );
};

export default BeerPage;
