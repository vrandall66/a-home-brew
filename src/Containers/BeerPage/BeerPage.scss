.BeerPage {
  background-color: #BAD9B5;
  display: flex;
  flex-direction: column;
  height: 69vh;
  margin: 0% 1.25%;
  margin-top: 6%;
  overflow: scroll;
  padding: 2% 0% 4% 0%;
  width: 78vw;
}

ul {
  list-style: none;
}

.BeerPage__div--right {
  display: flex;
}

.BeerPage__section--description {
  display: flex;
}

.BeerPage__div--description,
.BeerPage__div--brewersTips,
.BeerPage__div--pairsWith,
.BeerPage__aside--ingredients {
  margin: 1%;
  padding: 1%;
}

.BeerPage__aside--ingredients {
  align-content: center;
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 250px;
  justify-content: center;
  width: 100%;
}

.BeerPage__section--tips-pairs-with {
  padding-right: 2%;
}