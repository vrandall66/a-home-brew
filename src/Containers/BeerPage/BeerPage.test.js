import React from 'react';
import { shallow } from 'enzyme';
import BeerPage from './BeerPage';

describe('BeerPage', () => {
  let wrapper;
  let beerDetails = {
    id: 1,
    name: 'Buzz',
    tagline: 'A Real Bitter Experience.',
    first_brewed: '09/2007',
    food_pairing: [
      'Spicy chicken tikka masala',
      'Grilled chicken quesadilla',
      'Caramel toffee cake'
    ],
    description:
      'A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.',
    image_url: 'https://images.punkapi.com/v2/keg.png',
    ingredients: {
      hops: [
        { name: 'Fuggles', amount: {}, add: 'start', attribute: 'bitter' },
        { name: 'First Gold', amount: {}, add: 'start', attribute: 'bitter' },
        { name: 'Fuggles', amount: {}, add: 'middle', attribute: 'flavour' },
        { name: 'First Gold', amount: {}, add: 'middle', attribute: 'flavour' },
        { name: 'Cascade', amount: {}, add: 'end', attribute: 'flavour' }
      ],
      malt: [
        { name: 'Maris Otter Extra Pale', amount: {} },
        { name: 'Caramalt', amount: {} },
        { name: 'Munich', amount: {} }
      ],
      yeast: 'Wyeast 1056 - American Aleâ„¢'
    },
    method: {
      fermentation: {
        temp: { value: 19, unit: 'celsius' }
      },
      mash_temp: [{ duration: 75, temp: { value: 64, unit: 'celsius' } }]
    },
    abv: 4.5,
    ibu: 60,
    target_fg: 1010,
    target_og: 1044,
    ebc: 20,
    srm: 10,
    ph: 4.4,
    attenuation_level: 75,
    boil_volume: {
      unit: 'litres',
      value: 25
    },
    brewers_tips:
      'The earthy and floral aromas from the hops can be overpowering. Drop a little Cascade in at the end of the boil to lift the profile with a bit of citrus.',
    contributed_by: 'Sam Mason <samjbmason>',
    volume: { value: 20, unit: 'litres' }
  };

  beforeEach(() => {
    wrapper = shallow(<BeerPage beerDetails={beerDetails} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
