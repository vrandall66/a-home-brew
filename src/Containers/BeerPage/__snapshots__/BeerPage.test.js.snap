// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerPage should match the snapshot 1`] = `
<div
  className="BeerPage"
  key="1"
>
  <div
    className="BeerPage__div--left"
  >
    <section
      className="BeerPage__section--header"
    >
      <h2
        className="BeerPage__h2--beerName"
      >
        Buzz
      </h2>
      <h3>
        A Real Bitter Experience.
      </h3>
    </section>
    <section
      className="BeerPage__section--description"
    >
      <div
        className="BeerPage__div--description"
      >
        <h3>
          Description
        </h3>
        <p>
          A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.
        </p>
      </div>
      <section
        className="BeerPage__section--tips-pairs-with"
      >
        <div
          className="BeerPage__div--brewersTips"
        >
          <h3>
            Brewer's Tips
          </h3>
          <p>
            The earthy and floral aromas from the hops can be overpowering. Drop a little Cascade in at the end of the boil to lift the profile with a bit of citrus.
          </p>
        </div>
        <div
          className="BeerPage__div--pairsWith"
        >
          <h3>
            Pairs well with:
          </h3>
          <ul>
            <li
              key="0"
            >
              Spicy chicken tikka masala
            </li>
            <li
              key="1"
            >
              Grilled chicken quesadilla
            </li>
            <li
              key="2"
            >
              Caramel toffee cake
            </li>
          </ul>
        </div>
      </section>
    </section>
  </div>
  <div
    className="BeerPage__div--right"
  >
    <aside
      className="BeerPage__aside--ingredients"
    >
      <h4>
        Ingredients
      </h4>
      <ParentAccordion
        accordions={
          Array [
            <Accordion
              items={
                Array [
                  "Attribute: bitter",
                  "When to add it: start",
                  "undefined undefined",
                ]
              }
              title="Fuggles"
            />,
            <Accordion
              items={
                Array [
                  "Attribute: bitter",
                  "When to add it: start",
                  "undefined undefined",
                ]
              }
              title="First Gold"
            />,
            <Accordion
              items={
                Array [
                  "Attribute: flavour",
                  "When to add it: middle",
                  "undefined undefined",
                ]
              }
              title="Fuggles"
            />,
            <Accordion
              items={
                Array [
                  "Attribute: flavour",
                  "When to add it: middle",
                  "undefined undefined",
                ]
              }
              title="First Gold"
            />,
            <Accordion
              items={
                Array [
                  "Attribute: flavour",
                  "When to add it: end",
                  "undefined undefined",
                ]
              }
              title="Cascade"
            />,
          ]
        }
        className="BeerPage__ParentAccordion--ingredients"
        key="hops"
        title="Hops"
      />
      <ParentAccordion
        accordions={
          Array [
            <Accordion
              items={
                Array [
                  "undefined undefined",
                ]
              }
              title="Maris Otter Extra Pale"
            />,
            <Accordion
              items={
                Array [
                  "undefined undefined",
                ]
              }
              title="Caramalt"
            />,
            <Accordion
              items={
                Array [
                  "undefined undefined",
                ]
              }
              title="Munich"
            />,
          ]
        }
        className="BeerPage__ParentAccordion--ingredients"
        key="malts"
        title="Malts"
      />
      <p>
        Yeast: 
        Wyeast 1056 - American Aleâ„¢
      </p>
    </aside>
    <section
      className="BeerPage__section--stats"
    >
      <ul>
        <li>
          ABV: 
          4.5
        </li>
        <li>
          Attenuation Level: 
          75
        </li>
        <li>
          Boil volume: 
          25
           
          litres
        </li>
        <li>
          EBC: 
          20
        </li>
        <li>
          IBU: 
          60
        </li>
        <li>
          PH: 
          4.4
        </li>
        <li>
          SRM: 
          10
        </li>
        <li>
          Target FG: 
          1010
        </li>
        <li>
          Target OG: 
          1044
        </li>
        <li>
          Fermentation: 
          19
           
          celsius
        </li>
        <li>
          Total volume this makes: 
          20
           
          litres
        </li>
        <li>
          Mash temperature:
        </li>
        <li
          key="0"
        >
          64
           
          celsius
           
        </li>
      </ul>
    </section>
  </div>
</div>
`;
